spec/unit/type/file/content_uri_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'spec_helper'
+require 'uri'
+
+describe Puppet::Type.type(:file).attrclass(:content_uri) do
+  include PuppetSpec::Files
+
+  describe "#validate" do
+
+    let(:path) { tmpfile('file_content_uri_validate') }
+    let(:resource) { Puppet::Type.type(:file).new(:path => path) }
+
+    it "should fail if the URI is not a valid URI" do
+      URI.expects(:parse).with('not_a_uri').raises RuntimeError
+      expect(lambda { resource[:content_uri] = 'not_a_uri' }).to raise_error(Puppet::Error)
```

Comment:
You're expecting an exception, I think it might need to be `expect { resource[:content_uri] = 'not_uri' }`. But avoiding the lambda might fix it.